File: src/main/java/zielu/intellij/ui/GtFormUi.java
package zielu.intellij.ui;

import com.intellij.openapi.Disposable;
import javax.swing.JComponent;

public interface GtFormUi extends Disposable {
  void init();

  JComponent getContent();

  void afterStateSet();

  default void dispose() {
    // do nothing
  }
}

File: src/main/kotlin/zielu/intellij/ui/GtFormUi.kt
package zielu.intellij.ui

import com.intellij.openapi.Disposable
import javax.swing.JComponent

internal interface GtFormUi : Disposable {
  val content: JComponent

  fun init()

  fun afterStateSet()

  override fun dispose() {
    // do nothing
  }
}
